src/state/utils.ts

```diff
@@ -1,6 +1,7 @@
 import { Loadable } from 'recoil';
+import { BaseBuilder, buildGPX } from 'gpx-builder';
```

Comment:
In all honesty, I think this is something the backend will need to take care of. It's a pretty heavy thing for the frontend to take care of and comes at a whooping 18kb gzipped price.

Reply:
Makes sense!